includes/cli/class-amp-cli-validation-command.php

```diff
@@ -157,49 +109,15 @@ final class AMP_CLI_Validation_Command {
 	 * @throws Exception If an error happens.
 	 */
 	public function run( $args, $assoc_args ) {
-		$this->include_conditionals      = [];
-		$this->force_crawl_urls          = false;
-		$this->limit_type_validate_count = (int) $assoc_args[ self::LIMIT_URLS_ARGUMENT ];
-
-		/*
-		 * Handle the argument and flag passed to the command: --include and --force.
-		 * If the self::INCLUDE_ARGUMENT is present, force crawling or URLs.
-		 * The WP-CLI command should indicate which templates are crawled, not the /wp-admin options.
-		 */
-		if ( ! empty( $assoc_args[ self::INCLUDE_ARGUMENT ] ) ) {
-			$this->include_conditionals = explode( ',', $assoc_args[ self::INCLUDE_ARGUMENT ] );
-			$this->force_crawl_urls     = true;
-		} elseif ( isset( $assoc_args[ self::FLAG_NAME_FORCE_VALIDATION ] ) ) {
-			$this->force_crawl_urls = true;
-		}
-
-		// Handle special case for Legacy Reader mode.
-		if (
-			AMP_Theme_Support::READER_MODE_SLUG === AMP_Options_Manager::get_option( Option::THEME_SUPPORT )
-			&&
-			ReaderThemes::DEFAULT_READER_THEME === AMP_Options_Manager::get_option( Option::READER_THEME )
-		) {
-			$allowed_templates = [
-				'is_singular',
-			];
-			if ( 'page' === get_option( 'show_on_front' ) ) {
-				$allowed_templates[] = 'is_home';
-				$allowed_templates[] = 'is_front_page';
-			}
-
-			$disallowed_templates = array_diff( $this->include_conditionals, $allowed_templates );
-			if ( ! empty( $disallowed_templates ) ) {
-				WP_CLI::error( sprintf( 'Templates not supported in legacy Reader mode with current configuration: %s', implode( ',', $disallowed_templates ) ) );
-			}
+		$this->assoc_args = $assoc_args;
 
-			if ( empty( $this->include_conditionals ) ) {
-				$this->include_conditionals = $allowed_templates;
-			}
-		}
+		$validation_url_provider = $this->get_validation_url_provider();
+		$validation_provider     = $this->get_validation_provider();
 
-		$number_urls_to_crawl = $this->count_urls_to_validate();
+		$number_urls_to_crawl = count( $validation_url_provider->get_urls() );
 		if ( ! $number_urls_to_crawl ) {
-			if ( ! empty( $this->include_conditionals ) ) {
+			/** @phpstan-ignore-next-line */
```

Comment:
This shouldn't be needed. As I understand it you would have to stub the `Utils\get_flag_value()` method, which can be done by:

```diff
diff --git a/includes/cli/class-amp-cli-validation-command.php b/includes/cli/class-amp-cli-validation-command.php
--- a/includes/cli/class-amp-cli-validation-command.php	(revision 6c449c04cf81c7c1b7228e07ed881922a68802a7)
+++ b/includes/cli/class-amp-cli-validation-command.php	(date 1605052727456)
@@ -116,7 +116,6 @@
 
 		$number_urls_to_crawl = count( $validation_url_provider->get_urls() );
 		if ( ! $number_urls_to_crawl ) {
-			/** @phpstan-ignore-next-line */
 			if ( ! empty( Utils\get_flag_value( $this->assoc_args, self::INCLUDE_ARGUMENT, [] ) ) ) {
 				WP_CLI::error(
 					sprintf(

diff --git a/tests/php/static-analysis-stubs/wp-cli.php b/tests/php/static-analysis-stubs/wp-cli.php
--- a/tests/php/static-analysis-stubs/wp-cli.php	(revision 6c449c04cf81c7c1b7228e07ed881922a68802a7)
+++ b/tests/php/static-analysis-stubs/wp-cli.php	(date 1605053104533)
@@ -83,4 +83,23 @@
     function make_progress_bar($message, $count, $interval = 100)
     {
     }
+
+	/**
+	 * Return the flag value or, if it's not set, the $default value.
+	 *
+	 * Because flags can be negated (e.g. --no-quiet to negate --quiet), this
+	 * function provides a safer alternative to using
+	 * `isset( $assoc_args['quiet'] )` or similar.
+	 *
+	 * @access public
+	 * @category Input
+	 *
+	 * @param array  $assoc_args Arguments array.
+	 * @param string $flag       Flag to get the value.
+	 * @param mixed  $default    Default value for the flag. Default: NULL.
+	 * @return mixed
+	 */
+	function get_flag_value( $assoc_args, $flag, $default = null )
+	{
+	}
 }
```

Reply:
You may need to clear the `phpstan` result cache (`vendor/bin/phpstan clear-result-cache`) before running static analysis again.

Reply:
That seemed to work. updated in 6f42fd3